mcfly.bash

```diff
@@ -44,15 +43,11 @@ function mcfly_prompt_command {
     command tail -n100 "${HISTFILE}" >| "${MCFLY_HISTORY}"
   fi
 
-  history -a "${MCFLY_HISTORY}" # Append history to $MCFLY_HISTORY.
-  # Run mcfly with the saved code. It will:
-  # * append commands to $HISTFILE, (~/.bash_history by default)
-  #   for backwards compatibility and to load in new terminal sessions;
-  # * find the text of the last command in $MCFLY_HISTORY and save it to the database.
-  $MCFLY_PATH add --exit ${exit_code} --append-to-histfile
-  # Clear the in-memory history and reload it from $MCFLY_HISTORY
-  # (to remove instances of '#mcfly: ' from the local session history).
-  history -cr "${MCFLY_HISTORY}"
+  # Run mcfly with the last history entry on stdin. It will:
+  # * Append the command to $HISTFILE, (~/.bash_history by default)
+  # * Parse out the command and and save it to the database.
+  HISTTIMEFORMAT="%s:" history 1 | mcfly add --exit $exit_code --command-from-stdin
```

Comment:
Does this still work on bash <4?

Reply:
According to the bash release notes, `HISTTIMEFORMAT` was added in Bash 3, released in 2004. I did a quick test to be sure, it behaves the same as bash 4. Should be good.